src/Components/Components/src/Routing/LocationChangingContext.cs

```diff
@@ -38,13 +37,13 @@ internal LocationChangingContext(string targetLocation, string? historyEntryStat
     /// Gets a <see cref="System.Threading.CancellationToken"/> that can be used to determine if this navigation was canceled
     /// (for example, because the user has triggered a different navigation).
     /// </summary>
-    public CancellationToken CancellationToken => _cts.Token;
+    public CancellationToken CancellationToken { get; }
```

Comment:
I think we should consider passing the cancellation token as a separate argument to the handler, there is no relationship between the cancellation token and the rest of the data in this class.

In addition to that, it is "interesting" that this class looks "identical" to `LocationChangedEventArgs` except for the CancellationToken in this one (and the method to prevent the navigation). 

I am not suggesting that we "reuse" the same class, but an indicator that the cancellation token should probably not be here.

Reply:
@javiercn The challenge with having the `CancellationToken` as its own argument in the callback is that `EventCallback<TValue>` only allows there to be a single parameter, so if you define a location changing handler using `<NavigationLock OnBeforeInternalNavigation="Foo" />`, then `Foo()` can only accept a single argument (the `LocationChangingContext`). So the only option in that case is to leave the `CancellationToken` as a property on `LocationChangingContext`.

I guess one way we could make this work is that we have a separate `LocationChangingEventArgs` whose two properties are the `LocationChangingContext` and the `CancellationToken`. So, if you add a location changing handler using the `NavigationManager` API, your callback will still have a separate argument for the `CancellationToken`. I'm not sure how I feel about having to do `args.Context.PreventNavigation()` rather than just `context.PreventNavigation()` though.

Thoughts?

Reply:
And also, the limitations of `EventCallback<TValue>` are also a reason why having a `PreventNavigation()` method could be preferred over relying on a return value from the handler (since event callbacks cannot return anything other than a `Task`).